### PR TITLE
Editor user documentation template

### PR DESCRIPTION
**Pull request by Doralitze**
We need a user manual at some point. This would be the first step. Comments regarding the structure are appreciated.

### CODE REVIEW

File: Editor/internal/preprocessing.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+1. Resolve virtual filters
+2. Instantiation of constants for default parameters of filters
+3. Perform filter optimizations
+4. Pack the show file for fish
+
+## Resolving virtual filters
+TODO explain how virtual filter resolution works.
+
+## Placement of constants
+The GUI iterates over all left empty inputs of filters and creates constant filters for these ports. It does so by creating one constant per data type and value and reuses them for each occurance.
```
Comment (LRahlff) [inline Editor/internal/preprocessing.md]: there are multiple constant filter needed for each data_type, cause the default may differ

File: Editor/Filters/Readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Using filters
+
+This document describes the general ussage of the filter editor together with some special configuration editors for various more complex filters. For a general overview over available filters, please consult [this reference](Fish_Doku/Filter_Types.md).
```
Comment (LRahlff) [inline Editor/Filters/Readme.md]: The link does not work right now

**Comment by LRahlff** [comment]
i am a little bit confused because the folder structure does not follow the structure the links in the files. Should one should examine the files through the folders? I think maybe all files should be in (maybe only one) subdirectory(s) so one would first go through the readme.

**Comment by Doralitze** [comment]
> i am a little bit confused because the folder structure does not follow the structure the links in the files. Should one should examine the files through the folders? I think maybe all files should be in (maybe only one) subdirectory(s) so one would first go through the readme.

IMHO it is good to have these two structures as the links provided within the readme provide a quick-start-guide like experience whereas the folder structure prove a topic based navigation (for users that know what they're looking for).

**Review by LRahlff — APPROVED** [review]
Then it LGTM

File: Editor/Filters/Readme.md (hunk 2 above)
Comment (Doralitze) [reply Editor/Filters/Readme.md]: This has been fixed now.

File: Editor/internal/preprocessing.md (hunk 1 above)
Comment (Doralitze) [reply Editor/internal/preprocessing.md]: It has been made a bit more clear now.

**Pull request merged**